src/Core/Content/Test/Newsletter/SalesChannel/NewsletterSubscribeRouteTest.php

```diff
@@ -108,6 +108,71 @@ public function testSubscribe(): void
         static::assertSame(1, $count);
     }
 
+    public function testResubscribeAfterUnsubscribe(): void
+    {
+        // 1: register user at all, using double opt-in
+        $connection = $this->getContainer()->get(Connection::class);
+        $this->browser
+            ->request(
+                'POST',
+                '/store-api/newsletter/subscribe',
+                [
+                    'email' => 'test@test.de',
+                    'option' => 'subscribe',
+                    'storefrontUrl' => 'http://localhost',
+                ]
+            );
+
+        static::assertTrue($this->browser->getResponse()->isSuccessful());
+        $row = $connection->fetchAssoc('SELECT * FROM newsletter_recipient WHERE email = "test@test.de"');
+        static::assertSame('notSet', $row['status']);
+
+        // 2: confirm double opt-in
+        $this->browser
+            ->request(
+                'POST',
+                '/store-api/newsletter/confirm',
+                [
+                    'email' => 'test@test.de',
+                    'hash' => $row['hash'],
+                ]
+            );
+
+        static::assertTrue($this->browser->getResponse()->isSuccessful());
+        $row = $connection->fetchAssoc('SELECT * FROM newsletter_recipient WHERE email = "test@test.de"');
+        static::assertSame('optIn', $row['status']);
+
+        // 3: unsubscribe
+        $this->browser
+            ->request(
+                'POST',
+                '/store-api/newsletter/unsubscribe',
+                [
+                    'email' => 'test@test.de',
+                ]
+            );
+
+        static::assertTrue($this->browser->getResponse()->isSuccessful());
+        $row = $connection->fetchAssoc('SELECT * FROM newsletter_recipient WHERE email = "test@test.de"');
+        static::assertSame('optOut', $row['status']);
+
+        // 4: resubscribe
+        $this->browser
+            ->request(
+                'POST',
+                '/store-api/newsletter/subscribe',
+                [
+                    'email' => 'test@test.de',
+                    'option' => 'subscribe',
+                    'storefrontUrl' => 'http://localhost',
+                ]
+            );
+
+        static::assertTrue($this->browser->getResponse()->isSuccessful());
+        $row = $connection->fetchAssoc('SELECT * FROM newsletter_recipient WHERE email = "test@test.de"');
+        static::assertSame('notSet', $row['status']);
+    }
+
```

Comment:
That is a great idea, as this fails currently. The issue is, that `NewsletterConfirmRouter::confirm()` expects `confirmedAt` to be `null`.
So there probably is no other way than to reset the `confirmedAt` to null when unsubscribing.

Reply:
why not remove the check from the 'getBeforeConfirmSubscribeValidation'? It is private and prevents an new confirmation. (It would be better to keep the confirmedAt until someone subscribes again)

Reply:
All right. I removed it from the validation constraint collection and restored the previous version.
I also extended the test to check that the confirmation date is null / not null and that it changes after confirming a second time.